Modules/Devices/src/Devices.Domain/Entities/Identities/IdentityDeletionProcess.cs

```diff
@@ -120,10 +122,30 @@ public void GracePeriodReminder3Sent(IdentityAddress address)
 
     public void Approve(IdentityAddress address, DeviceId approvedByDevice)
     {
-        if (Status != DeletionProcessStatus.WaitingForApproval)
-            throw new DomainException(DomainErrors.NoDeletionProcessWithRequiredStatusExists());
+        EnsureDeletionProcessIsInWaitingForApprovalStatus();
 
         Approve(approvedByDevice);
         _auditLog.Add(IdentityDeletionProcessAuditLogEntry.ProcessApproved(Id, address, approvedByDevice));
     }
+
+    public void Reject(IdentityAddress address, DeviceId rejectedByDevice)
+    {
+        EnsureDeletionProcessIsInWaitingForApprovalStatus();
+
+        Reject(rejectedByDevice);
+        _auditLog.Add(IdentityDeletionProcessAuditLogEntry.ProcessRejected(Id, address, rejectedByDevice));
+    }
+
+    private void EnsureDeletionProcessIsInWaitingForApprovalStatus()
```

Comment:
Why does this method name have to be that specific? Also, the name doesn't have to contain the word "DeletionProcess", as it is defined on a class called "IdentityDeletionProcess". 
=> I would suggest `EnsureStatus(DeletionProcessStatus.WaitingForApproval)` (maybe you can find better wording, but you get the idea).